### PR TITLE
Wiggleplot

### PR DESCRIPTION
**Pull request by aissah**
Wiggle plot module data

### CODE REVIEW

**Comment by d-chambers** [comment]
Ready for review @aissah or are you still working on this?

**Comment by aissah** [comment]
> Ready for review @aissah or are you still working on this?

I am still working on it. Will probably finish it by noon.

**Comment by d-chambers** [comment]
Ok, no rush. You can just add the ready_for_review tag on right when you are done.

**Review by d-chambers — CHANGES_REQUESTED** [review]
Just a few small suggestions. Can you also post a screenshot of the output in this thread for future reference?

File: dascore/viz/wiggle.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+import matplotlib.pyplot as plt
+import numpy as np
+
+from dascore.constants import PatchType
+from dascore.utils.patch import patch_function
+from dascore.utils.plotting import _add_time_axis_label, _format_time_axis, _get_ax
+
+
+@patch_function()
```
[the remark this answers is not in the document]
Comment (aissah) [reply dascore/viz/wiggle.py:15]: The x-axis is always time but it doesn't require any specific input dimensions. Could you point out which line gives that impression?

File: dascore/viz/wiggle.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    dims = patch.dims
+    assert len(dims) == 2, "Can only make wiggle plot of 2D Patch"
+    patch = patch.transpose(..., dim)
+    data = patch.data
+    dims = patch.dims
+    dims_r = tuple(reversed(dims))
+    coords = {dimen: patch.coords[dimen] for dimen in dims}
+
+    if dim == "time":
+        maxOfTraces = abs(data).max(axis=1)
```
[the remark this answers is not in the document]
Comment (aissah) [reply dascore/viz/wiggle.py]: Ok, this song is a discovery. I have made the changes and will allow pep8 to guide my path in the future

File: dascore/viz/wiggle.py [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+                    dtype="timedelta64[ns]",
+                )
+                + axisAcrossWiggles[a]
+            )
+        else:
+            wiggle = (
+                intervals[min(a, totalWiggles - 2)] * data[a] + axisAcrossWiggles[a]
+            )
+        ax.plot(axisAlongWiggles, wiggle, color, alpha=1, linewidth=1)
+        wher = data[a] < 0
```
[the remark this answers is not in the document]
Comment (aissah) [reply dascore/viz/wiggle.py]: I am always somehow reluctant to use the name of an optional input as a variable name, makes me very anxious. But I will sacrifice myself for the greater good of dasdae

File: dascore/viz/wiggle.py [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+        The Patch object.
+    ax
+        A matplotlib object, if None create one.
+    color
+        Color of wiggles
+    timefmt
+        The format for the time axis.
+    show
+        If True, show the plot, else just return axis.
+    """
```
[the remark this answers is not in the document]
Comment (aissah) [reply dascore/viz/wiggle.py:37]: I am going to work on this but I might come by for some help next week

File: dascore/viz/wiggle.py [hunk 5]
```diff
@@ -0,0 +1,98 @@
+"""
+Module for wiggle plotting.
```
[the remark this answers is not in the document]
Comment (aissah) [reply dascore/viz/wiggle.py:2]: > 

wiggles_ss
wiggles_sstime

Did this work? The boring examples don't leave much to be seen

**Comment by d-chambers** [comment]
Hey @aissah 

[Here](https://dascore.org/contributing/documentation.html#doc-strings) is the section that explains docstrings. Let me know if it ins't clear and I while expand the section.

**Comment by d-chambers** [comment]
Rebased to master, should be good after that.

**Pull request merged**